sklearn/preprocessing/data.py

```diff
@@ -1179,9 +1180,39 @@ def powers_(self):
         combinations = self._combinations(self.n_input_features_, self.degree,
                                           self.interaction_only,
                                           self.include_bias)
-        return np.vstack(np.bincount(c, minlength=self.n_input_features_)
+        return np.vstack(bincount(c, minlength=self.n_input_features_)
                          for c in combinations)
 
+    def get_feature_names(self, input_features=None):
+        """
+        Return feature names for output features
+
+        Parameters
+        ----------
+        input_features : list of string, length n_features, optional
+            String names for input features if available. By default,
+            "x0", "x1", ... "xn_features" is used.
+
+        Returns
+        -------
+        output_feature_names : list of string, length n_output_features
+
+        """
+        powers = self.powers_
+        if input_features is None:
+            input_features = ['x%d' % i for i in range(len(powers))]
```

Comment:
Sorry for too many doubts, but it is mentioned that ``powers_[i, j] is the exponent of the jth input in the ith output.`` Shouldn't this mean that shape is ``(n_output_features, n_input_features)`` ? Also looking at ``powers_``, vstack stacks row-wise right and then there should be ``n_output_features`` number of rows.